exo-go/src/docker_helpers/index.go

```diff
@@ -0,0 +1,81 @@
+package dockerHelpers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Originate/exosphere/exo-go/src/process_helpers"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+)
+
+// RemoveDanglingImages removes all dangling images on the machine
+func RemoveDanglingImages(c *client.Client) error {
+	ctx := context.Background()
+	filtersArgs := filters.NewArgs()
+	filtersArgs.Add("dangling", "true")
+	imageSummaries, err := c.ImageList(ctx, types.ImageListOptions{
+		All:     false,
+		Filters: filtersArgs,
+	})
+	if err != nil {
+		return err
+	}
+	for _, imageSummary := range imageSummaries {
+		_, err = c.ImageRemove(ctx, imageSummary.ID, types.ImageRemoveOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RemoveDanglingVolumes removes all dangling volumes on the machine
+func RemoveDanglingVolumes(c *client.Client) error {
+	ctx := context.Background()
+	filtersArgs := filters.NewArgs()
+	filtersArgs.Add("dangling", "true")
+	volumesListOKBody, err := c.VolumeList(ctx, filtersArgs)
+	if err != nil {
+		return err
+	}
+	for _, volume := range volumesListOKBody.Volumes {
+		err = c.VolumeRemove(ctx, volume.Name, false)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// PullImage pulls the given image from DockerHub
+func PullImage(c *client.Client, image string, done func(error)) {
+	ctx := context.Background()
+	_, err := c.ImagePull(ctx, image, types.ImagePullOptions{})
+	done(err)
```

Comment:
Lets avoid callbacks unless absolutely necessary. Go is mostly a synchronous language and uses goroutines when we need to perform things in parallel. It also isn't needed in this case where you could just return an error.